src/lib/Libifl/pbsD_deljob.c

```diff
@@ -68,15 +70,17 @@
 int
 __pbs_deljob(int c, char *jobid, char *extend)
 {
-	struct attropl *aoplp = NULL;
+	char *list[2];
+	struct batch_deljob_status *res = NULL;
 
 	if ((jobid == NULL) || (*jobid == '\0'))
 		return (pbs_errno = PBSE_IVALREQ);
 
-	return PBSD_manager(c, PBS_BATCH_DeleteJob,
-		MGR_CMD_DELETE,
-		MGR_OBJ_JOB,
-		jobid,
-		aoplp,
-		extend);
+	list[0] = jobid;
+	list[1] = NULL;
+
+	res = __pbs_deljoblist(c, list, 1, extend);
```

Comment:
i think it's better to have only one codepath for job deletion. That way, any enhancements done to deljoblist can be used by deljob. If there's a different code path for job deletion then we should probably remove it.

Reply:
I would suggest removing any dead code created as part of making this change. You can defer it but both should happen together.
Multi-svr specific changes are already handled for pbs_deljob as it is calling PBSD_manager internally.

Reply:
deljoblist added another path to deleting jobs, so we should remove the path that exists via manager. This PR removes the client to server path. I think the server itself uses the manager path to talk to mom, removing that seems out of scope for this PR, we can take it up separately.

Reply:
There should be code associated with what you have removed from the client side in req_deletejob(). That becomes dead code as part of this change. If you are removing the client path, remove the same from the server as well.

If you are seeing it as out of scope, refrain from updating the client now and take it up together.

Reply:
the server still uses the old request type for various things, for example, it uses it to send delete job request to remote servers in case the job was moved, it uses it for preemption by deletion, for deleting jobs when a reservation ends, and to tell mom to delete a job, i might be missing other cases. So, certainly not dead code just because I changed the client to not use it. We should change the server to not use it, yes, but it's an internal refactoring which needs wide spanning changes in multiple daemons, completely out of scope for this PR.